### ISSUE TITLE
Breaking change from Next 15+: awaiting parameter promises

### ISSUE THREAD

**Issue by songyang-dev** [microsoft/vscode-typescript-next#143]
From Next 15 onwards, parameters in an asynchronous component needs to be awaited before being used. Here is an article explaining [this](https://fortifiedhq.com/blog/next-js-15-dynamic-routes-params-promise). After implementing the new breaking changes, the TS linter displays a warning.

```ts
export default async function BlogPostPage({ params }: Props) {
  const { slug } = await params; // 'await' has no effect on the type of this expression.ts(80007)
```

Not awaiting the parameter results in an error in the console only.

```
$ npm run dev
Error: Route "/blog/[slug]" used `params.slug`. `params` should be awaited before using its properties. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis
    at BlogPostPage (app/blog/[slug]/page.tsx:22:11)
  20 | export default async function BlogPostPage({ params }: Props) {
  21 |   // copy
> 22 |   const { slug } = params;
```

I suggest updating the TS extension to not display this warning.

**Comment by mjbvz** [comment]
This needs to be fixed in the next.js typings. Please file an issue against them about that
